### ISSUE TITLE
Coordinate system?

### ISSUE THREAD

**Issue by badicsalex** [nordmoen/eskf-rs#4]
TL;DR: Can you elaborate what the expected axes and IMU rotation handedness are in this project?

It seems like NED with right handed rotation, but I'm really not sure.

Also my guess is that it can work with any coordinate system as long as handedness is correct and gravity and orientation are initialized correctly.

--------

### Background

I'm creating high-frequency positioning data from GNSS and IMU data coming from an android phone that's in a phone holder in the car. The phone holder holds the phone at not so ideal angles, and the road itself is not perfectly flat. The gyros seem very precise, accelerometer not so much.

I've implemented an incredibly simple algorithm that assumes a fixed phone orientation, the car is not slipping and is traveling on perfectly flat ground. This way I can project any acceleration data to a single forward vector, use gyros to get the heading, and use the occassional GNSS fix to correct gyro and accelerometer errors. Biases and sensitivity are pre-calibrated.
While driving on actual flat ground, this works well.

Right now I need to extend this algorithm so that it self-calibrates for orientation and is able to handle somewhat sloped roads, so I'm moving to a Kalman Filter based solution, and this library seems very promising.

However, I cannot get it to work even with the data during on flat driving, and I suspect that the coordinate system conversion is wrong. Either the handedness is not OK, and the ESKF cannot represent the mirroring, or it just cannot converge at all if the orientation starts with something way out of whack.

### My default coordinate system

For the IMU is X: Right, Y: Up, Z: Forward, and the standstill measured acceleration vector in ideal orientation is (0,9.8,0), i.e. up.  The "global" coordinate system is a local tangent plane with X:east, Y: up, Z: north. Notice that these are both left-handed.

Gyroscope "coordinates" use the same axes and right-handed rotation.

**Comment by badicsalex** [comment]
It does indeed work if I use it with NED coordinates. This should probably be documented.

(I did have to introduce a way to separately set the initial variance of the orientation to fix the other problems I mentioned)

**Comment by badicsalex** [comment]
I did some experimentation, and now I'm pretty sure it works with all right-handed coordinate systems (I switched from NED to the common OpenGL one). I could not get left handed coordinate systems to work at all, no matter what I did with the gyroscope vector.

**Comment by badicsalex** [comment]
I did even more experimentation, and it turns out the lib works with left handed coordinate systems too, you just have to make sure that the gyro rotations are also left handed.
